src/components/encounter/encounter-form.component.tsx

```diff
@@ -725,6 +739,20 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
         setForm(form_temp);
       });
     }
+
+    if (referencedFields?.find((item) => item.dependantField === fieldName)) {
```

Comment:
@arodidev I don't understand what are trying to do here

Reply:
It checks whether the field with the changed value exists in the array of referenced fields, and if it does, it updates the value of the referenced field so that it is usable within the `select-concept-answers` datasource.